### ISSUE TITLE
regression: unrecognized arguments error when using subparsers and config file

### ISSUE THREAD

**Issue by kuba** [bw2/ConfigArgParse#26]
Test file:
```
#!/usr/bin/env python
import configargparse
parser = configargparse.ArgParser(args_for_setting_config_path=["-c"])
parser.add_subparsers().add_parser('foo')
parser.add_argument("-b", "--bar")
print parser.parse_args()
```

Test config:
```
bar = baz
```

Problem
```
(venv)jakub@iks:~/dev/letsencrypt/lets-encrypt-preview$ pip install git+https://github.com/bw2/ConfigArgParse.git@0.9.3
Downloading/unpacking git+https://github.com/bw2/ConfigArgParse.git@0.9.3
  Cloning https://github.com/bw2/ConfigArgParse.git (to 0.9.3) to /tmp/pip-80_Rgc-build
  Running setup.py (path:/tmp/pip-80_Rgc-build/setup.py) egg_info for package from git+https://github.com/bw2/ConfigArgParse.git@0.9.3
    
Installing collected packages: ConfigArgParse
  Running setup.py install for ConfigArgParse
    
Successfully installed ConfigArgParse
Cleaning up...
(venv)jakub@iks:~/dev/letsencrypt/lets-encrypt-preview$ ./t.py -c t.ini foo                                                                                                                         
Namespace(bar='baz', config_file='t.ini')
(venv)jakub@iks:~/dev/letsencrypt/lets-encrypt-preview$ pip uninstall ConfigArgParse                                                                                                                
Uninstalling ConfigArgParse:
  /home/jakub/dev/letsencrypt/lets-encrypt-preview/venv/lib/python2.7/site-packages/ConfigArgParse-0.9.3-py2.7.egg-info
  /home/jakub/dev/letsencrypt/lets-encrypt-preview/venv/lib/python2.7/site-packages/configargparse.py
  /home/jakub/dev/letsencrypt/lets-encrypt-preview/venv/lib/python2.7/site-packages/configargparse.pyc
Proceed (y/n)? y
  Successfully uninstalled ConfigArgParse
(venv)jakub@iks:~/dev/letsencrypt/lets-encrypt-preview$ pip install git+https://github.com/bw2/ConfigArgParse.git@e0463e0ec5fbd1f453ed833abebc633c77d4e0c6                                        
Downloading/unpacking git+https://github.com/bw2/ConfigArgParse.git@e0463e0ec5fbd1f453ed833abebc633c77d4e0c6
  Cloning https://github.com/bw2/ConfigArgParse.git (to e0463e0ec5fbd1f453ed833abebc633c77d4e0c6) to /tmp/pip-6fNLp7-build
  Could not find a tag or branch 'e0463e0ec5fbd1f453ed833abebc633c77d4e0c6', assuming commit.
  Running setup.py (path:/tmp/pip-6fNLp7-build/setup.py) egg_info for package from git+https://github.com/bw2/ConfigArgParse.git@e0463e0ec5fbd1f453ed833abebc633c77d4e0c6
    
Installing collected packages: ConfigArgParse
  Running setup.py install for ConfigArgParse
    
Successfully installed ConfigArgParse
Cleaning up...
(venv)jakub@iks:~/dev/letsencrypt/lets-encrypt-preview$ ./t.py -c t.ini foo
usage: t.py [-h] [-c CONFIG_FILE] [-b BAR] {foo} ...
t.py: error: unrecognized arguments: --bar baz
```

**Comment by kuba** [comment]
```
2f6faceae0199d64e2075e94daa3acf2375951b1 is the first bad commit
commit 2f6faceae0199d64e2075e94daa3acf2375951b1
Author: bw2 <weisburd@broadinstitute.org>
Date:   Wed May 13 00:53:10 2015 -0400

    Fix for #13 - copied SUPPRESS and other global constants from argparse to configargparse module namespace.
    Enhancement #9 - added constructor arg to add a 'dump config' option.
    Fix for #14 - refactored how unknown config file options are processed. Added 'ignore_unknown_config_file_keys' constructor arg as a better-named version of the 'allow_unknown_config_file_keys' arg which is now deprecated.
    Enhancement #15 - added 'auto_env_var_prefix' constructor arg.

:100644 100644 9936f518389b78134801dd1de4d99351241c6a04 f9c98c5eab6ef86382b1f05aaa1358d22bce7f8b M      configargparse.py
:040000 040000 da6e7f7526dad2ffc1a9b5f7cd3b38cf23104228 953dee718d966b531ff615568f9aff80d1142e3c M      tests
```

**Comment by Kedrigern** [comment]
+1

**Comment by kuba** [comment]
@bw2, could you please have a look at this?

**Comment by bw2** [comment]
Not yet sure how to fix this. Suggestions would be appreciated.

**Comment by bw2** [comment]
Thanks for tracking down the commit that broke this.
This will be fixed by PR #33 .
